llvm/lib/SYCLLowerIR/ESIMD/ESIMDRemoveHostCode.cpp

```diff
@@ -0,0 +1,71 @@
+//===-- ESIMDRemoveHostCode.cpp - remove host code for ESIMD -----------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// ESIMD code is not often run on the host, but we still compile for the host.
+// If requested by the user, remove the implementations of all ESIMD functions
+// to possibly speed up host compilation time.
+//===----------------------------------------------------------------------===//
+
+#define DEBUG_TYPE "ESIMDRemoveHostCodePass"
+
+#include "llvm/Demangle/Demangle.h"
+#include "llvm/Demangle/ItaniumDemangle.h"
+#include "llvm/IR/Function.h"
+#include "llvm/IR/Module.h"
+#include "llvm/Pass.h"
+#include "llvm/SYCLLowerIR/ESIMD/ESIMDUtils.h"
+#include "llvm/SYCLLowerIR/ESIMD/LowerESIMD.h"
+#include "llvm/Support/Debug.h"
+#include "llvm/TargetParser/Triple.h"
+
+using namespace llvm;
+using namespace llvm::esimd;
+namespace id = itanium_demangle;
+PreservedAnalyses ESIMDRemoveHostCodePass::run(Module &M,
+                                               ModuleAnalysisManager &) {
+  bool Modified = false;
+  assert(!Triple(M.getTargetTriple()).isSPIR() &&
+         "Pass should not be run for SPIR targets");
+  for (auto &F : M.functions()) {
```

Comment:
I guess we can skip function declarations right away, no need to demangle them

Reply:
Thanks, will skip decls and add those TODOs